GymBuddies/ViewControllers/RoutineDetailsViewController.m

```diff
@@ -85,6 +90,49 @@ -(void)setTopViewContent{
 }
 
 
+#pragma mark - Like methods
+
+-(void)setLikedState{
+    if(self.isLiked){
+        [self.likeButton setImage:[UIImage systemImageNamed:kLikedFilledRoutineButtonImage] forState:UIControlStateNormal];
+        self.likeButton.tintColor = [UIColor systemRedColor];
+    } else {
+        [self.likeButton setImage:[UIImage systemImageNamed:kLikedNormalRoutineButtonImage] forState:UIControlStateNormal];
+        self.likeButton.tintColor = [UIColor systemBlueColor];
+    }
+}
+
+
+- (IBAction)didTapLike:(id)sender {
+    [self likeAction];
+}
+
+
+-(void)likeAction{
+    if(self.isLiked){
+        [self.likeButton setImage:[UIImage systemImageNamed:kLikedNormalRoutineButtonImage] forState:UIControlStateNormal];
```

Comment:
duplication of logic. See suggestion for `likeAction` above. Also -- this method seems to be referenced in multiple places. You should then combine them into 1.